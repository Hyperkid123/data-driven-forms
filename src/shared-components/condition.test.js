import React from 'react';
import { mount } from 'enzyme';
import { Form } from 'react-final-form';
import Condition from './condition';
import toJson from 'enzyme-to-json';

describe('<Condition />', () => {
  const childId = 'condition-child';

  it('should not render children if condition is not met', () => {
    const wrapper = mount(
      <Form onSubmit={ jest.fn() }>
        { () => (
          <Condition when="foo" is="bar">
            <div id={ childId }></div>
          </Condition>
        ) }
      </Form>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render children if condition is met', () => {
    const wrapper = mount(
      <Form onSubmit={ jest.fn() } initialValues={{
        foo: 'bar',
      }}>
        { () => (
          <Condition when="foo" is="bar">
            <div id={ childId }></div>
          </Condition>
        ) }
      </Form>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should not render children if value is not in array', () => {
    const wrapper = mount(
      <Form onSubmit={ jest.fn() } initialValues={{
        foo: 'pox',
      }}>
        { () => (
          <Condition when="foo" is={ [ 'bar', 'fux', 'quax' ] }>
            <div id={ childId }></div>
          </Condition>
        ) }
      </Form>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render children if value is in array', () => {
    const wrapper = mount(
      <Form onSubmit={ jest.fn() } initialValues={{
        foo: 'bar',
      }}>
        { () => (
          <Condition when="foo" is={ [ 'bar', 'fux', 'quax' ] }>
            <div id={ childId }></div>
          </Condition>
        ) }
      </Form>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
