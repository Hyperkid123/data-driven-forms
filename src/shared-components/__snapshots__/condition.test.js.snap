// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Condition /> should not render children if condition is not met 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
>
  <Condition
    is="bar"
    when="foo"
  >
    <Field
      format={[Function]}
      name="foo"
      parse={[Function]}
      subscription={
        Object {
          "value": true,
        }
      }
    />
  </Condition>
</ReactFinalForm>
`;

exports[`<Condition /> should not render children if value is not in array 1`] = `
<ReactFinalForm
  initialValues={
    Object {
      "foo": "pox",
    }
  }
  onSubmit={[MockFunction]}
>
  <Condition
    is={
      Array [
        "bar",
        "fux",
        "quax",
      ]
    }
    when="foo"
  >
    <Field
      format={[Function]}
      name="foo"
      parse={[Function]}
      subscription={
        Object {
          "value": true,
        }
      }
    />
  </Condition>
</ReactFinalForm>
`;

exports[`<Condition /> should render children if condition is met 1`] = `
<ReactFinalForm
  initialValues={
    Object {
      "foo": "bar",
    }
  }
  onSubmit={[MockFunction]}
>
  <Condition
    is="bar"
    when="foo"
  >
    <Field
      format={[Function]}
      name="foo"
      parse={[Function]}
      subscription={
        Object {
          "value": true,
        }
      }
    >
      <div
        id="condition-child"
      />
    </Field>
  </Condition>
</ReactFinalForm>
`;

exports[`<Condition /> should render children if value is in array 1`] = `
<ReactFinalForm
  initialValues={
    Object {
      "foo": "bar",
    }
  }
  onSubmit={[MockFunction]}
>
  <Condition
    is={
      Array [
        "bar",
        "fux",
        "quax",
      ]
    }
    when="foo"
  >
    <Field
      format={[Function]}
      name="foo"
      parse={[Function]}
      subscription={
        Object {
          "value": true,
        }
      }
    >
      <div
        id="condition-child"
      />
    </Field>
  </Condition>
</ReactFinalForm>
`;
